truffle/src/main/java/org/jruby/truffle/language/parser/jruby/BodyTranslator.java

```diff
@@ -1827,6 +1828,12 @@ public RubyNode visitInstAsgnNode(org.jruby.ast.InstAsgnNode node) {
                 ret = new WriteInstanceVariableNode(context, sourceSection, name, self, IntegerCastNodeGen.create(context, sourceSection, rhs));
                 return addNewlineIfNeeded(node, ret);
             }
+        } else if (path.equals(corePath + "exception.rb")) {
+            if (name.equals("@errno")) {
```

Comment:
Could we just modify `exception.rb` in place to use a primitive to read the error, rather than having to rewrite here?